src/backends/tensorflow/tensorflow_backend_tf.cc

```diff
@@ -625,9 +625,61 @@ TRTISTF_ModelCreateFromGraphDef(
     if (node.op() == "Placeholder") {
       potential_inputs =
           TRTISTF_IOListNew(node.name().c_str(), nullptr, potential_inputs);
+
+      TRTISTF_IO* io = potential_inputs->io_;
+
+      TRTISTF_DataType dt;
+      tensorflow::DataType tf_dt;
+      std::vector<int64_t> tensor_shape;
+      auto attr_map = node.attr();
+      for (auto it=attr_map.begin(); it != attr_map.end(); it++) {
+          auto key = it->first;
+          auto value = it->second;
+          if (key == "dtype") {
+              dt = ConvertDataType(value.type());
+              tf_dt = value.type();
+          }
+          if (value.has_shape()) {
+              auto shape = value.shape();
+              for (int i=0; i<shape.dim_size(); ++i) {
+                  auto dim = shape.dim(i);
+                  auto dim_size = dim.size();
+                  tensor_shape.push_back(dim_size);
+              }
+          }
+      }
+
+      io->data_type_ = dt;
```

Comment:
What is dt is undefined? Should initialize it to invalid. Also, you don't ever seem to use this calculated type and shape. For savedmodel we collect these and then in savedmodel_backend.cc we use them to check against the configuration. Here you collect them but you didn't change graphdef_backend.cc to use them to check against the configuration. Also, need some way to represent unknown datatype or shape so that you know when to check and when you can't (for savedmodel we also assume there is type and shape but for shape at least that may change in future so will need this not just for graphdef).

Reply:
Was a part of earlier changes. remove. 

This was reverted because for GraphDef models the datatype stored is sometimes a valid datatype but it is not _correct_. 
For example, for the Resnet GraphDef model, the shape of input was [224,3] while the true shape expected is [224, 224, 3], similarly the datatype for the Softmax layer in some networks was **INT32** but it is actually supposed to be **FLOAT32**.